### ISSUE TITLE
`no_repeat_ngram_size` is accepted on `/v1/chat/completions` and then silently ignored

### ISSUE THREAD

**Issue by noonkho** [jundot/omlx#3741]
## Summary

`no_repeat_ngram_size` is accepted by the API — no 400, no warning — and has **no effect
whatsoever**. The same request with and without it returns byte-identical output, in the same
number of tokens, with the same `finish_reason`.

This matters because it is the knob `baidu/Unlimited-OCR`'s own official pipeline relies on: it
generates with `no_repeat_ngram_size=35` and `ngram_window=128`. Without it, the model falls into a
repeat loop on any near-empty page and burns the whole `max_tokens` budget.

So would suggest either implement it, or reject unknown sampling fields with a 400. Silently accepting a knob that
does nothing is the worst of the three, because a caller cannot tell the difference between "the
guard is on and this page still loops" and "the guard was never applied".

## Environment

| | |
|---|---|
| oMLX | 0.6.4 (the .dmg app) |
| Host | Apple Silicon Mac (M3 generation) |
| Model | `mlx-community/Unlimited-OCR-bf16` |

## Evidence

One page holding four short lines, `max_tokens: 1024`, `temperature: 0`:

| request | time | `completion_tokens` | `finish_reason` | distinct lines |
|---|---|---|---|---|
| baseline | 7.0 s | 1024 | `length` | 130 / 382 |
| `no_repeat_ngram_size: 35` | 3.7 s | 1024 | `length` | 130 / 382 |
| `no_repeat_ngram_size: 35`, `ngram_window: 128` | 3.7 s | 1024 | `length` | 130 / 382 |

The three answers are identical strings. The tail of each is the same line repeated to the cap:

```
<|det|>text [50, 154, 320, 172]<|/det|>To: The Registrar, High Court
<|det|>text [50, 154, 320, 172]<|/det|>To: The Registrar, High Court
<|det|>text [50, ...
```

With `no_repeat_ngram_size=35` honoured, that 35-token sequence could not repeat.

For contrast, the knobs that **are** implemented do change the result on the same page:

| request | time | `completion_tokens` | `finish_reason` |
|---|---|---|---|
| `frequency_penalty: 0.8` | 1.5 s | 340 | `stop` |
| `repetition_penalty: 1.15` | 0.9 s | 103 | `stop` |

So this is specific to `no_repeat_ngram_size`, not to penalties in general.

## Minimal reproduction

```python
import base64, json, urllib.request

BASE, KEY, MODEL = "http://YOUR-HOST:8000/v1", "YOUR-KEY", "Unlimited-OCR-bf16"

def ask(extra):
    img = base64.b64encode(open("near_empty_page.png", "rb").read()).decode()
    body = {"model": MODEL, "max_tokens": 1024, "temperature": 0,
            "messages": [{"role": "user", "content": [
                {"type": "text", "text": "Multi page parsing."},
                {"type": "image_url", "image_url": {"url": "data:image/png;base64," + img}}]}]}
    body.update(extra)
    req = urllib.request.Request(BASE + "/chat/completions", data=json.dumps(body).encode(),
        headers={"Content-Type": "application/json", "Authorization": f"Bearer {KEY}"})
    with urllib.request.urlopen(req, timeout=600) as r:
        return json.load(r)["choices"][0]["message"]["content"]

a = ask({})
b = ask({"no_repeat_ngram_size": 35})
print("identical:", a == b)          # True
```

Use any page with very little text on it — a separator page, a blank scan back, an exhibit divider.

## Expected

Either:

* `no_repeat_ngram_size` (and `ngram_window`) are applied as an n-gram logits processor, matching
  `transformers` and the model's official pipeline; or
* the request is rejected with a 400 naming the unsupported field.

## Why it matters

Document models are the models most likely to loop, because blank and near-blank pages are normal
in a real bundle: separator sheets, backs of scans, exhibit dividers. On this model, one such page
costs 29 s and 8,192 wasted tokens. The n-gram guard is the tool the model's authors chose for
exactly that, and today there is no way to reach it through oMLX.
